lib/src/components/NxLoadWrapper/__tests__/NxLoadWrapper.test.tsx

```diff
@@ -4,48 +4,87 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 import React from 'react';
-import * as enzymeUtils from '../../../__testutils__/enzymeUtils';
-import NxLoadWrapper, { Props } from '../NxLoadWrapper';
-import NxLoadError from '../../NxLoadError/NxLoadError';
-import NxLoadingSpinner from '../../NxLoadingSpinner/NxLoadingSpinner';
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
+
+import NxLoadWrapper from '../NxLoadWrapper';
 
 describe('NxLoadError', function() {
   const children = <div className="children"/>,
-      retryHandler = jest.fn(),
-      getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxLoadWrapper, { children, retryHandler });
+      retryHandler = jest.fn();
+  const renderEl = rtlRenderElement(NxLoadWrapper, {children, retryHandler});
+  const quickRender = rtlRender(NxLoadWrapper, {children, retryHandler});
 
   it('renders a NxLoadError if there is an error', function() {
-    const ErrorFixture = () => <NxLoadError error="foo" />;
-
-    expect(getShallowComponent({ error: 'foo' })).toMatchElement(<ErrorFixture/>);
-    expect(getShallowComponent({ error: 'foo', loading: true })).toMatchElement(<ErrorFixture />);
+    const componentWithoutError = renderEl();
+    const componentWithError = renderEl({error: 'Error!'});
+    expect(componentWithoutError).not.toHaveAttribute('role', 'alert');
+    expect(componentWithoutError?.textContent).toContain('');
+    expect(componentWithError).toBeInTheDocument();
+    expect(componentWithError).toHaveAttribute('role', 'alert');
+    expect(componentWithError?.textContent).toContain('Error!');
   });
 
-  it('passes the retryHandler to NxLoadError', function () {
-    expect(getShallowComponent({ error: 'foo' }).find(NxLoadError)).toHaveProp('retryHandler', retryHandler);
+  it('passes the retryHandler to component', function () {
```

Comment:
I recommend creating a `describe` block for "when there is an error", which would include the test that an alert is rendered, a test that a Retry button is rendered within the alert, and a test that the `retryHandler` is called when that button is clicked

Reply:
> I recommend creating a `describe` block for "when there is an error", which would include the test that an alert is rendered, a test that a Retry button is rendered within the alert, and a test that the `retryHandler` is called when that button is clicked. And also a test that the children content and loading spinner are _not_ rendered.
> 
> More broadly, I'd structure this whole file into separate describe blocks for the three big cases that this component handles: when there's an error, when it's loading, and when neither of those are the case and the children are getting rendered. Each of those describe blocks should have tests not only that the right things are rendered but also that the wrong things aren't rendered (for instance the loading tests should check that the error alert is not present).

Done 78a493683d4a2e4a515b7a25bbac20cebc8344ab